account_invoice_triple_discount/models/account_invoice.py

```diff
@@ -32,8 +32,15 @@ class AccountInvoiceLine(models.Model):
         default=0.0,
     )
 
-    @api.multi
-    @api.depends('discount2', 'discount3')
+    # Each iteration of api.multi is done into a different cache!
```

Comment:
I don't think thi is the expected behavior, so looks like it is a bug of odoo core.
You might open an issue in there addressing the wrong behavior or create a PR if you have actually found the misbehaving code.